tools/cli-client/internals/trigger/build_metadata.go

```diff
@@ -49,12 +49,16 @@ func InferBuildMetadata() BuildMetadata {
 	// use regex to match the repo location
 	regex := regexp.MustCompile("github.com[:/](canonical/.*).git")
 	matches := regex.FindStringSubmatch(remoteURL)
+	if len(matches) < 2 {
+		fmt.Fprintf(os.Stderr, "oci-factory must be called in a git local repository under organization [canonical]\n")
+		os.Exit(1)
+	}
 	source := matches[1]
```

Comment:
Then we should use `matches[0]` and `if len(matches) == 1` to match with this change.

Reply:
The length of a successfully matched group should be 2, since the first match group is the whole regex, and the second match group is the `(canonical/.*)`.